File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/ResolveInstanceHandler.py
from javonet.core.handler.AbstractCommandHandler import *
from javonet.core.referenceCache.ReferencesCache import ReferencesCache


class ResolveInstanceHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        try:
            if len(command.payload) != self._required_parameters_count:
                raise Exception("ResolveInstanceHandler parameters mismatch!")

            references_cache = ReferencesCache()
            return references_cache.resolve_reference(command)
        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Linux/X64/core/handler/resolve_instance_handler.rb
require_relative '../reference_cache/references_cache'
require_relative 'abstract_command_handler'

class ResolveInstanceHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end

  def process(command)
    return resolve_reference(command)
  end

  def resolve_reference(command)
    if command.payload.length != @required_parameters_count
      raise ArgumentError.new "Resolve Instance parameters mismatch"
    end
    begin
      references_cache = ReferencesCache.instance
      return references_cache.resolve_reference(command.payload[0])
    rescue Exception => ex
      return ex
    end
  end
end